extension/lib/login.js

```diff
@@ -0,0 +1,101 @@
+const BigCommerce = require('node-bigcommerce')
+const request = require('request-promise-native')
+const get = require('lodash.get')
+const InvalidCredentialsError = require('./errors/InvalidCredentialsError')
+const BigcommerceCustomerRepository = require('./respository/BigcommerceCustomerRepository')
+
+let customerRepo
+
+/**
+ * @param {PipelineContext} context
+ * @param {LoginInput} input
+ * @returns {Promise<LoginResponse>}
+ */
+module.exports = async (context, input) => {
+  if (!customerRepo) {
+    customerRepo = BigcommerceCustomerRepository.getInstance(new BigCommerce({
+      logLevel: 'info',
+      clientId: context.config.clientId,
+      accessToken: context.config.accessToken,
+      storeHash: context.config.storeHash,
+      responseType: 'json',
+      apiVersion: 'v2'
+    }))
+  }
+
+  try {
+    const success = await submitLoginForm(
+      `${input.shopUrl}/login.php?action=check_login`,
+      input.parameters.login,
+      input.parameters.password
+    )
+    if (!success) {
+      throw new InvalidCredentialsError()
+    }
+
+    const customer = await customerRepo.getCustomerByEmail(input.parameters.login)
```

Comment:
Please move out getting the customer in steps of getUser pipeline. Keep in mind it is not necessary to have get user implemented in this scope, but as you have put effort to it, I don't mind keeping the changes.

Reply:
Not really possible.
The getUser step we will need for the getUser pipeline will take an ID as input, but here we only have an email address as input.